mlir/include/Gradient/IR/GradientDialect.td

```diff
@@ -39,4 +41,19 @@ def Gradient_Dialect : Dialect {
 class Gradient_Op<string mnemonic, list<Trait> traits = []> :
         Op<Gradient_Dialect, mnemonic, traits>;
 
+//===----------------------------------------------------------------------===//
+// Gradient dialect types.
+//===----------------------------------------------------------------------===//
+
+class Gradient_Type<string name, string typeMnemonic, list<Trait> traits = []>
+        : TypeDef<Gradient_Dialect, name, traits> {
+    let mnemonic = typeMnemonic;
+}
+
+def ParameterVectorType : Gradient_Type<"ParameterVector", "vec"> {
+    let summary = "a parameter vector (dynamically resizable array)";
+    let parameters = (ins "Type":$elementType);
+    let assemblyFormat = "`<` $elementType `>`";
+}
```

Comment:
We definitely could! If we did, it seems appropriate to not have it in the `gradient` dialect because it's no longer specific to any gradient methods. Do you think it's worth the addition of a new dialect? Or would you recommend keeping it here?

Reply:
Yeah I don't think it needs to live in the gradient dialect either. A new dialect seems overkill, but there isn't really anywhere else for it to live 🤔 Something small that's just a namespace I guess.

Reply:
I moved the type and ops to a `catalyst` dialect meant to act as a sort of standard library. Happy to re-evaluate where to put them if you have additional thoughts 🙂